mm2src/coins/qrc20.rs

```diff
@@ -1083,10 +1100,15 @@ impl MarketCoinOps for Qrc20Coin {
         from_block: u64,
         _swap_contract_address: &Option<BytesJson>,
     ) -> TransactionFut {
```

Comment:
I think that the `MarketCoinOps::wait_for_tx_spend` method shouldn't return `TransactionErr` because it actually doesn't broadcast any transaction to the blockchain network.
I'd add another Transaction future that returns `String` error instead of `TransactionErr`.
It's only my opinion, so it's up to you to change or not

Reply:
I think `TransactionErr` fits better, because we have functions that don't return futures.

https://github.com/KomodoPlatform/atomicDEX-API/blob/9444edabe014b1d5519e1784159f9234e87f16ce/mm2src/coins/utxo/slp.rs#L412

I'd rather have `TransactionFut = Box<dyn Future<Item = TransactionEnum, Error = String>>` and `TxRecoverableFut = Box<dyn Future<Item = TransactionEnum, Error = TransactionFutErr>>`.
What do you think?

Reply:
But on the other hand, it may lead to confusion in macros, because `try_tx_fus` will actually return `TxRecoverableFut`. So, as I said, it's up to you :)